src/Cli/ConfigGenerator.cs

```diff
@@ -36,23 +36,38 @@ public class ConfigGenerator
         /// </summary>
         public static bool TryGenerateConfig(InitOptions options, RuntimeConfigLoader loader, IFileSystem fileSystem)
         {
-            if (!TryGetConfigFileBasedOnCliPrecedence(loader, options.Config, out string runtimeConfigFile))
+            string runtimeConfigFile = RuntimeConfigLoader.DEFAULT_CONFIG_FILE_NAME;
+            if (!string.IsNullOrWhiteSpace(options.Config))
             {
-                runtimeConfigFile = RuntimeConfigLoader.DEFAULT_CONFIG_FILE_NAME;
-                _logger.LogInformation($"Creating a new config file: {runtimeConfigFile}");
+                _logger.LogInformation("Generating user provided config file with name: {configFileName}", options.Config);
+                runtimeConfigFile = options.Config;
+            }
+            else
+            {
+                string? environmentValue = Environment.GetEnvironmentVariable(RuntimeConfigLoader.RUNTIME_ENVIRONMENT_VAR_NAME);
+                if (!string.IsNullOrWhiteSpace(environmentValue))
+                {
+                    _logger.LogInformation("The environment variable {variableName} has a value of {variableValue}", RuntimeConfigLoader.RUNTIME_ENVIRONMENT_VAR_NAME, environmentValue);
+                    runtimeConfigFile = RuntimeConfigLoader.GetEnvironmentFileName(RuntimeConfigLoader.CONFIGFILE_NAME, environmentValue);
+                    _logger.LogInformation("Generating environment config file: {configPath}", Path.GetFullPath(runtimeConfigFile));
+                }
+                else
+                {
+                    _logger.LogInformation("Generating default config file: {config}", Path.GetFullPath(runtimeConfigFile));
+                }
             }
 
             // File existence checked to avoid overwriting the existing configuration.
             if (fileSystem.File.Exists(runtimeConfigFile))
             {
-                _logger.LogError("Config file: {runtimeConfigFile} already exists. Please provide a different name or remove the existing config file.", runtimeConfigFile);
+                _logger.LogError("Config file: {runtimeConfigFile} already exists. Please provide a different name or remove the existing config file.",
+                    Path.GetFullPath(runtimeConfigFile));
```

Comment:
What if Path.GetFullPath throws exception?

Reply:
We should be using `fileSystem.Path` not `Path` directly to ensure we're working against the right file system.